examples/zoom-provider/server/zoomprovider.js

```diff
@@ -0,0 +1,123 @@
+const request = require('request')
+// const { data } = require('autoprefixer')
+const BASE_URL = 'https://zoom.us/v2'
+
+/**
+ * an example of a custom provider module. It implements @uppy/companion's Provider interface
+ */
+class ZoomProvider {
+  constructor (options) {
+    this.authProvider = ZoomProvider.authProvider
+  }
+
+  static get authProvider () {
+    return 'zoom'
+  }
+
+  // /users/{userId}/recordings
+  list ({ token, query }, done) {
+    const path = '/users/me/recordings'
```

Comment:
I'm noticing that zoom defaults to only returning content from the last day unless there's a longer date range specified (up to 1 month). Do any of the other providers have similar restrictions / how is this handled? Is there also a limit with Uppy on the number of items we can return / display? Zoom sets this at 300 but that seems like a lot?

Reply:
There's a [cursor query](https://github.com/transloadit/uppy/blob/a3809fcd2a5dce3e192cbbdeb18e2de90614b1a3/packages/%40uppy/companion/src/server/provider/onedrive/index.js#L34) which Companion uses for pagination of provider file list. It is often set in the [nextPagePath](https://github.com/transloadit/uppy/blob/a3809fcd2a5dce3e192cbbdeb18e2de90614b1a3/packages/%40uppy/companion/src/server/provider/onedrive/index.js#L111) field (something like `zoom/list/<folder ID if exists>?cursor=<cursor value>`).

Maybe for the case of zoom we can set the cursor value to the date range?

Reply:
Hi @ifedapoolarewaju I was looking at the cursor query, it seems that when I have extra fields on the cursor (eg as a nested object) it seems to cause requests to be triggered when I scroll up and down, whereas when I add some things as a query string and keep the cursor query as a single string, it doesn't do this.  Do you know why this might be?

Reply:
this is fixed, I just combined using an extra query field with the cursor query param